basic/file_test_image_cached_test.go

```diff
@@ -19,39 +20,59 @@ func TestFileCached(t *testing.T) {
 		t.Fatalf("Error creating executor: %v", err)
 	}
 
-	const numberOfInstances = 10 // Define how many instances you want to create
+	const numberOfInstances = 10
 	instances := make([]*knuu.Instance, numberOfInstances)
+	errs := make(chan error, numberOfInstances)
 
 	for i := 0; i < numberOfInstances; i++ {
-		instanceName := fmt.Sprintf("web%d", i+1) // Generates a unique name for each instance
+		instanceName := fmt.Sprintf("web%d", i+1)
 		instance, err := knuu.NewInstance(instanceName)
 		if err != nil {
-			t.Fatalf("Error creating instance '%v': %v", instanceName, err)
+			errs <- fmt.Errorf("Error creating instance '%v': %v", instanceName, err)
+			return
 		}
-
 		err = instance.SetImage("docker.io/nginx:latest")
 		if err != nil {
-			t.Fatalf("Error setting image for '%v': %v", instanceName, err)
+			errs <- fmt.Errorf("Error setting image for '%v': %v", instanceName, err)
+			return
 		}
```

Comment:
[question]: why do we need to use a channel slice for errors while the loop does not use a goroutin in it?

Reply:
I see. As we return right after the error and the execution is ended, we can use e.g. `t.Fatal` so no need for channels here.